### PR TITLE
Add initial WebHDFS itests

### LINKED ISSUE #314: Add integration tests for the new WebHDFS adapter
**Describe the task**
The new WebHDFS adapter needs integration tests to verify the functionality.

**Additional context**

### PR DESCRIPTION
**Pull request by mdang8**
#### What does this PR do?

Adds initial integration tests for the WebHDFS adapter. Currently, it has tests for the `isAvailable()` and `createResource()` adapter methods.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?

@cjlange 
@aaronilovici 
@clockard 

#### How should this be tested? (List steps with links to updated documentation)

Run the test class in Intellij or the command line and verify that all tests pass.

#### Any background context you want to provide?

These tests are for #311. Utilizes this library for creating Hadoop mini clusters: https://github.com/sakserv/hadoop-mini-clusters.

#### What are the relevant tickets?
Fixes #314 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: adapters/webhdfs-adapter/src/test/java/com/connexta/replication/adapters/webhdfs/WebHdfsClientTest.java
Comment (cjlange) [inline adapters/webhdfs-adapter/src/test/java/com/connexta/replication/adapters/webhdfs/WebHdfsClientTest.java:98]: I recall you saying that the HDFS cluster was restarting before every test, despite being located in the `@BeforeClass` block, but without that background knowledge, others may be concerned that the cluster isn't being restarted after this test.

**Review by cjlange — COMMENTED** [review]
No major observations.  Some of the tests do feel very similar to existing unit tests, but I'll leave it to your judgment whether you want to remove those.  For integration tests in general, I'd be most concerned with testing the interactions between our code and the external HDFS instance.

File: adapters/webhdfs-adapter/src/test/java/com/connexta/replication/adapters/webhdfs/WebHdfsClientTest.java
Comment (cjlange) [inline adapters/webhdfs-adapter/src/test/java/com/connexta/replication/adapters/webhdfs/WebHdfsClientTest.java:14]: Add the copyright to this file.

**Review by cjlange — CHANGES_REQUESTED** [review]

**Review by aaronilovici — APPROVED** [review]
Aside from the comment from Chris, this looks fantastic. A great breadth of use cases. Approved pending a comment explaining the point Chris raised.

File: adapters/webhdfs-adapter/src/test/java/com/connexta/replication/adapters/webhdfs/WebHdfsClientTest.java
Comment (mdang8) [reply adapters/webhdfs-adapter/src/test/java/com/connexta/replication/adapters/webhdfs/WebHdfsClientTest.java:98]: I can add a comment explaining that behavior.

**Review by cjlange — APPROVED** [review]

File: adapters/webhdfs-adapter/src/test/java/com/connexta/replication/adapters/webhdfs/WebHdfsClientTest.java
Comment (josephthweatt) [inline adapters/webhdfs-adapter/src/test/java/com/connexta/replication/adapters/webhdfs/WebHdfsClientTest.java:127]: Can't we just use `LOGGER.error` here?

Comment (mdang8) [reply adapters/webhdfs-adapter/src/test/java/com/connexta/replication/adapters/webhdfs/WebHdfsClientTest.java:127]: Ahh maybe that could've been worded better. I meant that currently there's no easy way to tell when a specific exception gets thrown because it ends up getting caught in a top-level method and a message gets printed to the logger instead. So the only way I could think of seeing the error was to track every error message that gets printed.

Comment (cjlange) [reply adapters/webhdfs-adapter/src/test/java/com/connexta/replication/adapters/webhdfs/WebHdfsClientTest.java:127]: There is a way to capture Logger messages during tests, but I need to do some more research around it.  We can revisit these tests when we have a clear approach.

**Review by josephthweatt — APPROVED** [review]

**Comment by mdang8** [comment]
build now

**Pull request merged**
